src/gcp/cloudfunctions.ts

```diff
@@ -20,17 +33,126 @@ interface Operation {
   error?: { code: number; message: string };
 }
 
-export const API_VERSION = "v1";
+export interface HttpsTrigger {
+  // output only
+  url?: string;
+  securityLevel?: SecurityLevel;
+}
 
-export const DEFAULT_PUBLIC_POLICY = {
-  version: 3,
-  bindings: [
-    {
-      role: "roles/cloudfunctions.invoker",
-      members: ["allUsers"],
-    },
-  ],
-};
+export interface EventTrigger {
+  eventType: string;
+  resource: string;
+  service?: string;
+  failurePolicy?: FailurePolicy;
+}
+
+export interface CorsPolicy {
+  allowOrigin: string[];
+  allowMethods?: string[];
+  allowHeaders?: string[];
+  exposeHeaders?: string[];
+}
+
+export interface SecretEnvVar {
+  key: string;
+  projectId: string;
+  secret: string;
+  version: string;
+}
+
+export interface SecretVolume {
+  mountPath: string;
+  projectId: string;
+  secret: string;
+  versions: {
+    version: string;
+    path: string;
+  }[];
+}
+
+export type Runtime = "nodejs6" | "nodejs8" | "nodejs10" | "nodejs12" | "nodejs14";
+export type CloudFunctionStatus =
+  | "ACTIVE"
+  | "OFFLINE"
+  | "DEPLOY_IN_PROGRESS"
+  | "DELETE_IN_PROGRESS"
+  | "UNKNOWN";
+export type SecurityLevel = "SECURE_ALWAYS" | "SECURE_OPTIONAL";
+
+export interface FailurePolicy {
+  // oneof action
```

Comment:
I see that you're still holding out hope for more options here 😄

Reply:
I later defined a `FunctionSpec` type and just simplify it to `retry` but I want to document the proto accurately in the GCP folder.